File: backend/app/schemas/itinerary.py
from typing import Any, Dict, Optional

from pydantic import BaseModel, Field


class ItineraryCreate(BaseModel):
    starting_latitude: Optional[float] = None
    starting_longitude: Optional[float] = None

    start_time: Optional[str] = None

    available_time_minutes: Optional[int] = None

    budget: Optional[int] = None

    interests: Optional[Dict[str, Any]] = Field(
        default=None
    )

    crowd_tolerance: Optional[float] = None

    itinerary: Optional[Dict[str, Any]] = Field(
        default=None
    )


class ItineraryResponse(BaseModel):
    id: str

    starting_latitude: Optional[float] = None
    starting_longitude: Optional[float] = None

    start_time: Optional[str] = None

    available_time_minutes: Optional[int] = None

    budget: Optional[int] = None

    interests: Optional[Dict[str, Any]] = None

    crowd_tolerance: Optional[float] = None

    itinerary: Optional[Dict[str, Any]] = None

    created_at: Optional[str] = None

File: frontend/src/api/itinerary.ts
import { API_BASE_URL } from './config';

export interface ItineraryRequest {
  starting_latitude: number;
  starting_longitude: number;
  start_time: string;
  available_time_minutes: number;
  budget: number;
  interests: Record<string, any> | string[];
  crowd_tolerance: number;
  itinerary?: Record<string, any>;
}

export async function createItinerary(
  data: ItineraryRequest
) {
  const response = await fetch(
    `${API_BASE_URL}/api/itineraries`,
    {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify(data),
    }
  );

  if (!response.ok) {
    throw new Error(
      `Failed to create itinerary: ${response.status}`
    );
  }

  return response.json();
}